### ISSUE TITLE
False-positive tag name detection

### ISSUE THREAD

**Issue by AleksandrHovhannisyan** [Esri/arcgis-html-sanitizer#95]
Test case:

```js
expect(sanitizer.sanitize("<0 or >1")).toBe("&lt;0 or &gt;1");
```

Result:

```
 ● Sanitizer › strip ignore tag

    expect(received).toBe(expected) // Object.is equality

    Expected: "&lt;0 or &gt;1"
    Received: "1"

      574 |     expect(sanitizer.sanitize(a)).toBe(aExpected);
      575 |     expect(sanitizer.sanitize(b)).toBe(bExpected);
    > 576 |     expect(sanitizer.sanitize("<0 or >1")).toBe("&lt;0 or &gt;1");
```

It seems the sanitizer greedily matches tag names by searching for any characters between a `<` and `>`, but it doesn't consider whether the contents in between actually correspond to a legal HTML tag name.

This is causing a bug where text containing simple numerical comparisons is getting stripped out.

Possibly related, I found that changing this line from `true` to `false` does not change the outcome of the existing tests, suggesting it's not actually doing anything:

https://github.com/Esri/arcgis-html-sanitizer/blob/dc79102c64f1d5804120e7d8d44caeb63bb92f36/src/index.test.ts#L567

**Comment by AleksandrHovhannisyan** [comment]
It seems this might be an issue with https://github.com/leizongmin/js-xss, as our `sanitize` method has no logic specific to matching tag names.

**Comment by AleksandrHovhannisyan** [comment]
Thinking on this some more... Maybe we just have to live with this behavior. Because you can't possibly account for the innumerable custom element tag names people are using. The sanitizer is just seeing `<0 or >1` as a tag name of `0` followed by an `or` attribute, followed by plaintext of `1` 😞

**Comment by BlakeStearman** [comment]
@AleksandrHovhannisyan Appreciate your comments and thoughts on this. I think this does boil down to the `sanitize` method is meant to strip tags and attributes that are not on the whitelist, and then encoding everything else. I will have to check into the `stripIgnoreTag` flag further, but that test case was originally added when the behavior of js-xss changed a default behavior in version 1.0.14 and we had to rollback to 1.0.13 and this was intended to verify behavior of 1.0.13.

I think what you might be wanting to use is the `encodeHTML` method, which we added in recent versions. That method will handle simple escaping `& < > \" ' /` within a string without stripping any potential HTML tags.
